### ISSUE TITLE
Support `change` in migrations as a shortcut for `up` and `down`.

### ISSUE THREAD

**Issue by hugopeixoto** [laravel/ideas#2334]
In most of the migrations I end up writing, `down` ends up being predictable by looking at `up`. For example:
- `Schema::create` becomes `Schema::dropIfExists`
- `Schema::table('users', function (Blueprint $table) { $table->string('email'); });` becomes `Schema::table('users', function (Blueprint $table) { $table->dropColumn('email'); });`

In these cases, Laravel could provide a different API that would remove this redundancy, making the migration creation process a bit easier.

I gave it a go at implementing this for creating and dropping tables. The fact that the migration objects are completely detached from the schema objects proved to be a problem. Here's my hacky approach so that you get the general idea of what I'm proposing.

Before, a migration that creates a table would look like this (ignoring comments and imports):

```php
class CreateUsersTable extends Migration {
  public function up() {
    Schema::create('users', function (Blueprint $table) {
      $table->id();
      $table->timestamps();
    });
  }

  public function down() {
    Schema::dropIfExists('users');
  }
}
```

Using `change`, it could look something like this:

```php
class CreateUsersTable extends NewMigration {
  public function change() {
    $this->create('users', function (Blueprint $table) {
      $table->id();
      $table->timestamps();
    });
  }
}
```

I used `$this->` instead of `Schema::` so that I can resolve `create` to either `Schema::create` or `Schema::dropIfExists` depending on the migration direction.

This is how I wrote `NewMigration` to add support for this:

```php
class NewMigration extends Migration
{
  public function create(string $table_name, $callback) {
    if ($this->direction == "up") {
      Schema::create($table_name, $callback);
    } else {
      Schema::dropIfExists($table_name);
    }
  }

  public function up() {
    $this->direction = "up";
    $this->change();
  }

  public function down() {
    $this->direction = "down";
    $this->change();
  }
}
```

It won't be this simple to implement reversibility onto every schema method. Not every case can be trivially reversible, so falling back to `up` and `down` would still be possible (and we don't need to break compatibility).

What are your thoughts on this?

**Comment by rtconner** [comment]
There are too many edge cases that would make this insanely complex to build. Indexes, db inserts and the like.

**Comment by hugopeixoto** [comment]
> There are too many edge cases that would make this insanely complex to build. Indexes, db inserts and the like.

Yeah, it would not be possible to build for every migration, since you can just write arbitrary code there.
That's what I meant with:

> Not every case can be trivially reversible, so falling back to up and down would still be possible

For these edge cases, you'd use `up` and `down`. But for trivial scenarios, you'd use `change`.
